### PR TITLE
PA-484: TrackProgress - use WireMock for testing APIs

### PR DESCRIPTION
**Pull request by benagain**

### CODE REVIEW

File: src/TrackProgress/SFA.DAS.TrackProgress.Api.AcceptanceTests/ApiTests/CallApi.cs [hunk 1]
```diff
@@ -77,681 +55,260 @@ public async Task Track_progress_with_single_matching_apprenticeship_and_no_sand
...
+        factory
+            .WithApprenticeship(apprenticeship)
+            .WithCourse(course);
+
+        client.DefaultRequestHeaders.Add(SubscriptionHeaderConstants.ForSandboxMode, isSandbox);
+
+        var response = await client.PostAsync(
+            $"/apprenticeships/{1}/{"2020-01"}/progress", validDto);
+
+        // TODO Ensure an update is called only when it is expected
```
Comment (Pauljgraham) [inline src/TrackProgress/SFA.DAS.TrackProgress.Api.AcceptanceTests/ApiTests/CallApi.cs:71]: We weren't able to test if a call was or wasn't made to the inner API is it worthwhile doing that test now?

File: src/TrackProgress/SFA.DAS.TrackProgress.Api.AcceptanceTests/testsettings.json [hunk 2]
```diff
@@ -3,14 +3,5 @@
...
-  "CommitmentsV2InnerApi": {
-    "url": "https://google.com/"
-  },
-  "trackProgressApi": {
-    "url": "https://google.co.uk/"
-  },
-  "coursesApi": {
-    "url": "https://google.co.uk/"
-  }
+  "APPINSIGHTS_INSTRUMENTATIONKEY": ""
```
Comment (Pauljgraham) [inline src/TrackProgress/SFA.DAS.TrackProgress.Api.AcceptanceTests/testsettings.json:6]: I don't think this testsettings.json file is needed. I removed it and everything ran fine. It seems to duplicate settings elsewhere

**Review by Pauljgraham — COMMENTED** [review]
A couple of very minor comments. Looks good

File: src/TrackProgress/SFA.DAS.TrackProgress.Api.AcceptanceTests/ApiTests/CallApi.cs (hunk 1 above)
Comment (benagain) [reply src/TrackProgress/SFA.DAS.TrackProgress.Api.AcceptanceTests/ApiTests/CallApi.cs:71]: Removed

File: src/TrackProgress/SFA.DAS.TrackProgress.Api.AcceptanceTests/testsettings.json (hunk 2 above)
Comment (benagain) [reply src/TrackProgress/SFA.DAS.TrackProgress.Api.AcceptanceTests/testsettings.json]: Removed

**Review by Chazzysb — APPROVED** [review]
Approved

**Review by Chazzysb — APPROVED** [review]
Approved

**Pull request merged**